### ISSUE TITLE
FirebaseMessaging does not support senderId or secondary app initialization, as the documentation suggests

### ISSUE THREAD

**Issue by Manuito83** [firebase/firebase-android-sdk#3987]
### Describe your environment

  * Platform: Android & Flutter
  * Firebase Component: Messaging & Documentation
  * Component version: n/a

### Describe the problem

According to the [documentation](https://firebase.google.com/docs/cloud-messaging/concept-options#receiving-messages-from-multiple-senders) it should be possible to receive messages from multiple senders in Firebase Messaging.

> _To enable this feature, make sure you have each sender's sender ID. When requesting registration, the client app fetches the token multiple times, each time with a different sender ID in audience field, using the token retrieval method for the given platform_

This was indeed possible some time ago, by specifying the sender(s) id(s) when getting the token, but it was [deprecated](https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceId#public-string-gettoken-string-senderid,-string-scope) and the current [getToken()](https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessaging#public-taskstring-gettoken) method does not accept a senderId parameter any longer.

There several other issues opened/closed referring to this problem, but in my opinion there have been no real fixes proposed yet.

Another [issue ](https://github.com/xamarin/GooglePlayServicesComponents/issues/438 )stated that there is a GetInstance missing parameter, but was closed with a Stackoverflow answer, containing some information from the Firebase team, that does not apply any longer.

https://github.com/firebase/firebase-android-sdk/issues/2588 and https://github.com/firebase/firebase-android-sdk/issues/3205 were closed as duplicates of https://github.com/firebase/firebase-android-sdk/issues/2285. But this last issue is a feature request more than an issue, while in my opinion it should be the opposite. In the conversation there is a solution proposed that involves retrieving a FirebaseMessaging instance, which is not documented anywhere.

As a response to https://github.com/firebase/firebase-android-sdk/issues/2778, it was stated that creating a new FirebaseApp instance with FirebaseOptions for the project is the only current supported way of handling multiple project. But this comes with the catch that FirebaseMessaging [does not allow](https://github.com/firebase/firebase-android-sdk/issues/2285) to be initialized with a secondary FirebaseApp, so I don't think it would be possible to add more than one project to use FirebaseMessaging, as the documentation suggest with the example of the news aggregator.

So unless there is something I'm not taking into account, I wanted to open this as a new issue to summarize all the information I've been able to gather.

So, basically we have a documentation that suggest a feature that doesn't seem to be longer possible (as it was deprecated), while there is no support for secondary apps in FirebaseMessaging. To be honest, I'm not sure how a news aggregator would be able to setup multiple messaging sources right now (at least as per the documentation).

**Comment by argzdev** [comment]
Hi @Manuito83, thank you for compiling all these information together! We appreciate the effort that went into this writing. This does sound like a use case which falls between a feature request and an issue.

I'll try and escalate this to our engineering team. However, given that they also have other tasks and issues to fulfill, it my take some time for them to dig deeper into this. That being said, while I’m unable to promise any timeline for this, if others are also in the same predicament, adding a thumbs up on this post can help our engineers prioritize this based on the demand and severity. Thanks!

**Comment by Manuito83** [comment]
Thanks a lot for the reply @argzdev.

The reason why I called this more an "issue" than a "feature request", is because this was indeed a feature that was used in many projects but that was deprecated a year ago with no other official workaround. This feature certainly helped in managing notifications from different sources, and by retiring it without notice many projects could have been affected. The fact that the documentation hasn't been changed for a year and a half makes me question whether the deprecation of this feature was really intended (i.e. Firebase decided not to officially support this any longer, for whatever reason), or if it was an unintended consequence of a change in the SDK (which I would classify as an "issue").

I'd like to point out, also, that while there is a non-documented workaround, as explained above...

```
val app = Firebase.initialize(this, options, "ANY_FIXED_STRING_EXCEPT_DEFAULT");
val firebaseMessaging = app.get(FirebaseMessaging::class.java) as FirebaseMessaging;
FirebaseMessaging.token.addOnCompleteListener{
    // Get token here
}
```

... this issue is causing some problems downstream... such as in some other Firebase implementations that depend on the Android SDK (e.g. Flutterfire), that had to [retire](https://pub.dev/packages/firebase_messaging/changelog#1000) support for this functionality completely (they have [no workaround](https://github.com/firebase/flutterfire/issues/7362)).

**Comment by argzdev** [comment]
Hi @Manuito83, sorry for the delayed response here. I've inquired with our engineers, it seems that multi-sender id is no longer supported after migrating to FIS, and there's currently no plans at this time to support multi-sender. So I don't think there's any official workaround atm.

However, thank you for pointing out these details. We will update our documentations accordingly, to not confuse our developers with a feature that no longer works.

That being said, I'll close this issue for now to avoid duplicates, since #2285 is active.
